File: python/twoSum.py
from ast import If
from operator import le


class Solution:
    def twoSum(self, nums, target):
        pointer1 = 0
        while (pointer1 < len(nums)):
            staticVal = nums[pointer1]
            pointer2 = pointer1+1
            while (pointer2 < len(nums)):
                if (staticVal+nums[pointer2] == target):
                    return [pointer1, pointer2]
                else:
                    pointer2 += 1
            pointer1 += 1


x = Solution()
nums = [3, 3]
target = 6
y = x.twoSum(nums, target)
print(y)

# Array sum
# https://leetcode.com/problems/two-sum/

File: typeScript/twoSum.ts
function twoSum(nums: number[], target: number): number[] | void {
  let pointer1: number = 0
  while (pointer1 < nums.length) {
    let pointer2: number = pointer1 + 1
    while (pointer2 < nums.length) {
      if (nums[pointer1] + nums[pointer2] == target) {
        return [pointer1, pointer2]
      } else {
        pointer2 += 1
      }
    }
    pointer1 += 1
  }
}

let nums: number[] = [3, 2, 4]
let target: number = 6
let x: number[] | void = twoSum(nums, target)
console.log(x)

// Error
// 'number' only refers to a type, but is being used as a value here.
// Leetcode question number 1
